### ISSUE TITLE
obb的性能疑惑

### ISSUE THREAD

**Issue by fakerhbj** [ultralytics/ultralytics#7590]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

obb的性能是采用了多尺度训练和多尺度测试出来的性能结果吗？

**Comment by Laughing-q** [comment]
@fakerhbj 是的

**Comment by kuazhangxiaoai** [comment]
我弄了一个yolov8 obb的项目，看上去还可以，请看一下：https://github.com/kuazhangxiaoai/yolov8-obb.git
![yolov8-obb](https://github.com/ultralytics/ultralytics/assets/54108453/90f2a648-1e60-4a83-8a4b-08f6f5ad2a0c)

**Comment by Laughing-q** [comment]
> > @fakerhbj 指标是多尺度训练多尺度测试的，速度性能是单张单尺度1024x1024的速度
> 
> 请问如何开启多尺度训练和多尺度预测呢？有配置文件或者相关文档吗？

因为DOTA数据集的分辨率很高，所以DOTA的训练是先采用滑窗的方式切割图片为1024x1024大小，这里的多尺度是指滑窗切割的时候的多尺度，具体可参考：https://docs.ultralytics.com/datasets/obb/dota-v2/#split-dota-images， 送入模型之前都还是采用统一的1024大小

**Comment by QibangLi** [comment]
> 我弄了一个yolov8 obb的项目，看上去还可以，请看一下：https://github.com/kuazhangxiaoai/yolov8-obb.git ![yolov8-obb](https://private-user-images.githubusercontent.com/54108453/296956976-90f2a648-1e60-4a83-8a4b-08f6f5ad2a0c.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0OTE0NTgsIm5iZiI6MTcwNTQ5MTE1OCwicGF0aCI6Ii81NDEwODQ1My8yOTY5NTY5NzYtOTBmMmE2NDgtMWU2MC00YTgzLThhNGItMDhmNmY1YWQyYTBjLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDExMzIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzMGU1NGZjOTE5ZjU2ZWFiZGVjMGZiMWE3OTE3YmU4MDg4YTJkYWYwNThlOWRlNWYxZDc3ODMwMWNjZGZjOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EtvcpNKfEOFuP_YSs7msX2WhSy79v1r59g16JUlk5Hs)

你好你这是yolov8s模型吗，同时想问一下，如何使用yolov8推理测试集，然后提交dota官网评测

**Comment by glenn-jocher** [comment]
@QibangLi 你好！很高兴你在使用YOLOv8进行OBB项目。关于使用YOLOv8进行推理和提交到DOTA官网评测的步骤，你可以参考我们的文档中的Predict和Val模式说明。这些文档会指导你如何进行模型推理和验证。祝你的项目成功！🌟

**Comment by mzy369** [comment]
> > > @fakerhbj 指标是多尺度训练多尺度测试的，速度性能是单张单尺度1024x1024的速度
> > 
> > 
> > 请问如何开启多尺度训练和多尺度预测呢？有配置文件或者相关文档吗？
> 
> 因为DOTA数据集的分辨率很高，所以DOTA的训练是先采用滑窗的方式切割图片为1024x1024大小，这里的多尺度是指滑窗切割的时候的多尺度，具体可参考：[https://docs.ultralytics.com/datasets/obb/dota-v2/#split-dota-images，](https://docs.ultralytics.com/datasets/obb/dota-v2/#split-dota-images%EF%BC%8C) 送入模型之前都还是采用统一的1024大小

请问数据集多尺度切割之后，还需要将multi_scale这个参数设置为True吗？

**Comment by Laughing-q** [comment]
@mzy369 不需要，针对DOTA数据的多尺度都是指切图的多尺度，训练都是1024x1024的固定分辨率

**Comment by puppetmonkey** [comment]
请问想要复现
yolov8 obb（dotav1）的结果（yolov8n-obb,map test50 :78.0 ），1. 我该使用哪一个超参数配置文件 ？还是默认的这个路径吗？‘ultralytics/cfg/
/default.yaml’   2.需要使用哪个预训练权重？是yolov8n.pt（coco）吗？ @Laughing-q

**Comment by Laughing-q** [comment]
@puppetmonkey 你可以通过如下的方式查看训练的参数：
```python
import torch

ckpt = torch.load("weights/yolov8m-obb.pt")
print(ckpt["train_args"])
```
所有的obb模型都是从头开始训练的，没有使用预训练加载

**Comment by glenn-jocher** [comment]
@LeiaJ520 hey there! 😊 It looks like you're interested in using YOLOv8 for inference on a test set and submitting results to the DOTA website. If you're working with a custom project like yolov8-obb, you'd typically follow these steps for inference and submission:

1. **Inference**: Use the `predict` mode with your model. Here's a quick example if you're using the CLI:
   ```bash
   yolo detect predict model=path/to/your/yolov8-obb.pt source=path/to/test/images
   ```
   Make sure to replace `path/to/your/yolov8-obb.pt` and `path/to/test/images` with the actual paths.

2. **Prepare Submission**: Format your results according to the DOTA evaluation guidelines. This usually involves converting your model's output to the DOTA submission format.

3. **Submit**: Follow the submission instructions on the DOTA competition website.

If you're looking for more specific guidance on formatting the results or the submission process, the DOTA competition guidelines or forums might have more detailed instructions. Good luck with your submission! 🚀

**Comment by STITCHESHHH** [comment]
> @mzy369 不需要，针对DOTA数据的多尺度都是指切图的多尺度，训练都是1024x1024的固定分辨率
您好，为什么我切分出的是682×682的分辨率，而不是1024×1024

**Comment by csh313** [comment]
> > @mzy369 不需要，针对DOTA数据的多尺度都是指切图的多尺度，训练都是1024x1024的固定分辨率
> > 您好，为什么我切分出的是682×682的分辨率，而不是1024×1024

你好，请问你解决了吗

**Comment by LeiaJ520** [comment]
还没有，多尺度里面可以指定切割的大小，可是我的代码一直报错

**Comment by csh313** [comment]
> 还没有，多尺度里面可以指定切割的大小，可是我的代码一直报错

就用官方的代码split_dota就可以吧，再用
from ultralytics.data.split_dota import split_trainval, split_test

# split train and val set, with labels.
split_trainval(
    data_root='path/to/DOTAv1.0/',
    save_dir='path/to/DOTAv1.0-split/',
    rates=[0.5, 1.0, 1.5],    # multiscale
    gap=500
)
# split test set, without labels.
split_test(
    data_root='path/to/DOTAv1.0/',
    save_dir='path/to/DOTAv1.0-split/',
    rates=[0.5, 1.0, 1.5],    # multiscale
    gap=500
)
这段分割，我用这个把v1分割成10w张，分辨率也是682，跑一下明显提高了map，就是数据太多了跑了两天。又从dota官网下载的，分割成了1024。

**Comment by LeiaJ520** [comment]
我也是用的这个代码，可能是我数据集结构有问题吧，总是报错

<img width="1115" alt="微信图片_20240513105430" src="https://github.com/ultralytics/ultralytics/assets/95119550/18c2d03e-f739-49bb-a921-3555dc323fa0">
另外，我还有个问题想请问下，就是在进行多尺度测试的时候，图片已经被划分为多个尺度的，那最后如何让整合检测结果，并提交到DOTA服务器上计算结果呢

**Comment by kuazhangxiaoai** [comment]
@csh313 你可以关注一下https://github.com/CAPTAIN-WHU/DOTA_devkit

**Comment by puppetmonkey** [comment]
> @puppetmonkey 你可以通过如下的方式查看训练的参数：
> 
> ```python
> import torch
> 
> ckpt = torch.load("weights/yolov8m-obb.pt")
> print(ckpt["train_args"])
> ```
> 
> 所有的obb模型都是从头开始训练的，没有使用预训练加载

但是按照您给的方法查看yolov8n-obb.pt训练参数，显示的'pretrained': True。想再次咨询一下obb 的预训练权重用的哪一个？

**Comment by puppetmonkey** [comment]
![image](https://github.com/ultralytics/ultralytics/assets/81809745/4ee99efc-a8a0-4e88-93d1-67b5e6a0642b)

**Comment by glenn-jocher** [comment]
你好，关于多尺度测试结果的整合，通常需要将检测结果进行后处理，比如合并重叠的检测框，然后按照DOTA的提交格式进行整理。具体的实现可以参考DOTA的官方工具或文档。
